### PR TITLE
[Merged by Bors] - feat: port Algebra.Category.Ring.Basic

### PR DESCRIPTION
**Pull request by kim-em**
This is a replacement for https://github.com/leanprover-community/mathlib4/pull/3105 which makes use of the refactors proposed in https://github.com/leanprover-community/mathlib4/pull/3900.

I'm happy to just merge this into #3105 if that's preferred, but since it depends on a refactor I thought I'd keep it separate for now.

- [x] depends on: #3900

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3900
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by jcommelin** [comment]
Here's the same diff, but rebased on top of master: https://github.com/leanprover-community/mathlib4/compare/master...RingCat2

File: Mathlib/Algebra/Category/Ring/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+  Bundled Ring
+set_option linter.uppercaseLean3 false in
+#align Ring RingCat
+
+namespace RingCat
+
+instance : BundledHom.ParentProjection @Ring.toSemiring :=
+  ⟨⟩
+
+-- Porting note: Another place where mathlib4 had derived a concrete category
```
Comment (jcommelin) [inline Mathlib/Algebra/Category/Ring/Basic.lean:143]: Do you really mean mathlib4?

**Review by jcommelin — COMMENTED** [review]
One comment, otherwise LGTM

bors d+

File: Mathlib/Algebra/Category/Ring/Basic.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Algebra/Category/Ring/Basic.lean:143]:
```suggestion
-- Porting note: Another place where mathlib had derived a concrete category
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**